src/CMakeLists.txt

```diff
@@ -131,9 +139,10 @@ function(jemalloc_config_run)
                   OUTPUT_VARIABLE output
                   OUTPUT_STRIP_TRAILING_WHITESPACE
                   COMMAND_ERROR_IS_FATAL ANY)
-  string(REGEX REPLACE ".*--with-jemalloc-prefix=([^ ]*).*" "\\1" JEMALLOC_PREFIX ${output})
-  if(${JEMALLOC_PREFIX} STREQUAL "")
-    set(${JEMALLOC_PREFIX} "${NONE}")
+  string(REGEX MATCH "--with-jemalloc-prefix=[^ ]*" JEMALLOC_PREFIX "${output}")
```

Comment:
Thanks for this... it seems like clearly an improvement/defensive in nature, at the very least.

Again for my own education I ask how you came across the desire to do this. Is it that if one builds jemalloc without explicitly setting this prefix thing to blank when configuring it, then the jemalloc-config output will omit --with-jemalloc-prefix= entirely?

(That may or may not be related to a certain problem someone found when setting the prefix to <none>. But don't worry about that... my thing to track down. Bottom line is you may have fixed here the problem they found-- I am just not sure yet.)

Reply:
> Is it that if one builds jemalloc without explicitly setting this prefix thing to blank when configuring it, then the jemalloc-config output will omit --with-jemalloc-prefix= entirely?

Yes!

Distros using RPM (Fedora and OpenSUSE) ship `jemalloc-devel` with `jemalloc-config` script, whose output contains no `--with-jemalloc-prefix`.

Distros using DEB (Debian and Ubuntu) ship `libjemalloc-dev` without `jemalloc-config` script at all.

On Debian, I built `jemalloc` myself (without setting any configuration options) to provided `jemalloc-config` and `jemalloc-config --config` outputs a blank line.

Actually `jemalloc-config --config` outputs arguments provided to `./configure ...` as is. And usually the `JEMALLOC_PERFIX` is not set to utilize `LD_PRELOAD` functionality.